check_pgactivity

```diff
@@ -6547,6 +6550,114 @@ sub check_pg_dump_backup {
     return status_ok( $me, [], \@perfdata );
 }
 
+=item B<pgss_dealloc>
+
+Check number of pg_stat_statements dealloc.
+
+Many dealloc can have performance impact. This could be due to:
+
+Application performing many different queries:
+
+* ORM
+
+* different utility statements (SET application_name...). In this case,
+  try to disable pg_stat_statements.track_utility.
+
+pg_stat_statements.max is too low
+
+This service uses the status file (see C<--status-file> parameter).
+
+The C<--critical> and C<--warning> thresholds are optional.
+Warning or critical will be raised if dealloc rate are above those thresholds.
+
+Perfdata returns the dealloc rate since last call.
+
+Required privileges: unprivileged role able to log in database where
+pg_stat_statements extensions has been created.
+
+=cut
+
+sub check_pg_stat_statements_dealloc {
+
+    my @rs;
+    my @perfdata;
+    my @perf_limits;
+    my @hosts;
+    my %dealloc;
+    my %new_dealloc;
+    my $dealloc_rate;
+    my $crit;
+    my $warn;
+    my %args    = %{ $_[0] };
+    my $me      = 'POSTGRES_PG_STAT_STATEMENTS_DEALLOC';
+    my $sql     = q{
+    SELECT
+        floor(extract(EPOCH from now())),
+        dealloc
+    FROM pg_stat_statements_info
+    };
+
+    @hosts = @{ parse_hosts %args };
+
+    pod2usage(
+    -message => 'FATAL: you must give only one host with service "commit_ratio".',
+    -exitval => 127
+    ) if @hosts != 1;
+
+    is_compat $hosts[0], 'check_pg_stat_statements_dealloc', $PG_VERSION_140 or exit 1;
```

Comment:
Note that having postgres 14 doesn't mean that you updated the pg_stat_statements extension to get the needed field.

Reply:
Indeed, I will add a test to check pgss' version.

Reply:
I added several tests:
* pg_stat_statements version must be above or equal 1.9
* pg_stat_statements has been created on target database
* pg_stat_statements has been loaded in shared_preload_libraries